Sources/SwiftProtobuf/Message+FieldMask.swift

```diff
@@ -0,0 +1,126 @@
+// Sources/SwiftProtobuf/Message+FieldMask.swift - Message field mask extensions
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extend the Message types with FieldMask utilities.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Message {
+
+  /// Checks whether the given path is valid for Message type.
+  ///
+  /// - Parameter path: Path to be checked
+  /// - Returns: Boolean determines path is valid.
+  public static func isPathValid(
+    _ path: String
+  ) -> Bool {
+    var message = Self()
+    return message.hasPath(path: path)
+  }
+
+  internal mutating func isPathValid(
+    _ path: String
+  ) -> Bool {
+    hasPath(path: path)
+  }
+}
+
+/// Defines available options for merging two messages.
+public struct MergeOption {
+
+  public init(replaceRepeatedFields: Bool = false) {
```

Comment:
I think the other review is debating if we want to list all the options on `init`, or have folks just manually set them after making an instance. In anycase, we should be consistent.

Reply:
My initial suggestion would just be:

```swift

extension Google_Protobuf_FieldMask {
  public struct MergeOptions {
    // ...
  }
}
```

That way it is (hopefully) clear that it is specific to FieldMask.

Reply:
Done